plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/DependencyResolver.java

```diff
@@ -0,0 +1,139 @@
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyFilter;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.wagon.WagonTransporterFactory;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.filter.DependencyFilterUtils;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.embedder.ICallable;
+import org.eclipse.m2e.core.embedder.IMavenExecutionContext;
+
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+// import org.eclipse.aether.transport.file.FileTransporterFactory;
+// import org.eclipse.aether.transport.http.HttpTransporterFactory;
+
+
+public class DependencyResolver {
+
+  public static List<String> getTransitiveDependencies(
+      String groupId, String artifactId, String version) throws DependencyResolutionException, CoreException {
+       
+    Artifact artifact = new DefaultArtifact(groupId + ":" + artifactId + ":" + version);
+
+    DependencyFilter filter = DependencyFilterUtils.classpathFilter(JavaScopes.COMPILE);
+
+    CollectRequest collectRequest = new CollectRequest();
+    collectRequest.setRoot( new Dependency(artifact, JavaScopes.COMPILE));
+    
+    final RepositorySystem system = newRepositorySystem();
+    // RepositorySystem system = MavenPluginActivator.getDefault().getRepositorySystem();
+    collectRequest.setRepositories(newRepositories(system));
+    final DependencyRequest request = new DependencyRequest(collectRequest, filter);
+
+    IMavenExecutionContext context = MavenPlugin.getMaven().createExecutionContext();
+    
+    ICallable<List<String>> callable = new ICallable<List<String>>() {
+      @Override
+      public List<String> call(IMavenExecutionContext context, IProgressMonitor monitor)
+          throws CoreException {
+        List<String> dependencies = new ArrayList<>();
+        RepositorySystemSession session = context.getRepositorySession();
+        try {
+          List<ArtifactResult> artifacts =
+              system.resolveDependencies(session, request).getArtifactResults();
```

Comment:
I changed this to move most of the setup into the callable, and to use the `MavenPluginActivator` and it works:
```patch
+        DependencyFilter filter = DependencyFilterUtils.classpathFilter(JavaScopes.COMPILE);
+        RepositorySystem system = MavenPluginActivator.getDefault().getRepositorySystem();
+        
+        CollectRequest collectRequest = new CollectRequest();
+        collectRequest.setRoot( new Dependency(artifact, JavaScopes.COMPILE));
+        collectRequest.setRepositories(newRepositories(system));
+        final DependencyRequest request = new DependencyRequest(collectRequest, filter);
```

Reply:
I still get exceptions. Can you push your changes on a different branch so I can see them? (same branch messes up the CLA check.)

Reply:
Pushed up as https://github.com/GoogleCloudPlatform/google-cloud-eclipse/tree/transitive-ii

Reply:
Thanks. I still see a failure in the test on your branch:

```
java.lang.NullPointerException
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:361)
	at com.google.cloud.tools.eclipse.util.DependencyResolver$1.call(DependencyResolver.java:81)
	at com.google.cloud.tools.eclipse.util.DependencyResolver$1.call(DependencyResolver.java:1)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:176)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
	at com.google.cloud.tools.eclipse.util.DependencyResolver.getTransitiveDependencies(DependencyResolver.java:92)
	at com.google.cloud.tools.eclipse.util.DependencyResolverTest.testResolve(DependencyResolverTest.java:30)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.pde.internal.junit.runtime.RemotePluginTestRunner.main(RemotePluginTestRunner.java:62)
	at org.eclipse.pde.internal.junit.runtime.PlatformUITestHarness$1.run(PlatformUITestHarness.java:47)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:35)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4024)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3700)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:694)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:606)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.runApp(NonUIThreadTestApplication.java:54)
	at org.eclipse.pde.internal.junit.runtime.UITestApplication.runApp(UITestApplication.java:47)
	at org.eclipse.pde.internal.junit.runtime.NonUIThreadTestApplication.start(NonUIThreadTestApplication.java:48)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1488)
```

I wonder what's different between our environments?

Reply:
That is odd as the [test succeeds on Travis too](https://travis-ci.org/GoogleCloudPlatform/google-cloud-eclipse/builds/269427408?utm_source=github_status&utm_medium=notification).

Reply:
Wonder if it has something to do with local maven repo, local maven version, or Eclipse version?

Reply:
I think that code is failing [here](https://github.com/apache/maven-resolver/blob/1.0.x/aether-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultRepositorySystem.java#L361), which suggests that either `result` is null or `result.getRoot()` is null, even though it should be set from the code above.

Are you running the test with all bundles enabled?

Reply:
Juts normal "Run as JUnit Plugin Test"

Let me push up your changes and see what happens on Travis.

Reply:
I think it has to be result.getRoot that's null. I don't see how result can be null there.